### ISSUE TITLE
OpenSSL fails to build

### ISSUE THREAD

**Issue by willyd** [Microsoft/vcpkg#407]
OpenSSL fails to build. I get the following error messages:
```
> vcpkg install openssl:x64-windows
-- CURRENT_INSTALLED_DIR=D:/Work/lib/vcpkg/installed/x64-windows
-- DOWNLOADS=D:/Work/lib/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=D:/Work/lib/vcpkg/packages/openssl_x64-windows
-- CURRENT_BUILDTREES_DIR=D:/Work/lib/vcpkg/buildtrees/openssl
-- CURRENT_PORT_DIR=d:/Work/lib/vcpkg/ports/openssl/.
-- Using cached D:/Work/lib/vcpkg/downloads/openssl-1.0.2j.tar.gz
-- Testing integrity of cached file...
-- Testing integrity of cached file... OK
-- Configuring x64-windows-rel
-- Configuring x64-windows-rel done
-- Configuring x64-windows-dbg
-- Configuring x64-windows-dbg done
-- Build x64-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:14 (message):
    Command failed: C:/Program Files/CMake/bin/cmake.exe;--build;.
    Working Directory: D:/Work/lib/vcpkg/buildtrees/openssl/x64-windows-rel
    See logs for more information:
      D:\Work\lib\vcpkg\buildtrees\openssl\build-x64-windows-rel-out.log
      D:\Work\lib\vcpkg\buildtrees\openssl\build-x64-windows-rel-err.log

Call Stack (most recent call first):
  ports/openssl/portfile.cmake:42 (vcpkg_execute_required_process)
  scripts/ports.cmake:73 (include)


Error: building package openssl:x64-windows failed
Please ensure sure you're using the latest portfiles with `vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: openssl:x64-windows
  Vcpkg version: 0.0.61-79399923b6cd98b9e77020615e433ef0560d5dc2

Additionally, attach any relevant sections from the log files above.
```
[Error Log](http://pastebin.com/ku29p2fZ)
[Output Log](http://pastebin.com/vReYwUa0)

Any ideas on what's wrong here?

**Comment by willyd** [comment]
Turns out I ran the install command from a shell where the `PATH` was set so that the build was picking msys2 perl and msys2 git. Maybe this has something to do with this. Once I used a cmd prompt without mys2 being in the `PATH` the build went well.

**Comment by ras0219-msft** [comment]
Thanks for figuring that out. It was probably perl causing the issue, if I had to guess.

We chose to allow pulling perl from the outside environment to potentially reduce the download pressure on users, but it may not be worth the cost of occasionally failing builds like you just experienced.

**Comment by j-forster** [comment]
I am having the same problems with this [error log](http://pastebin.com/7yVfHDSz).
It looks like perl is causing the problem.
What did you do to fix `PATH`? I cant find any msys2 perl preset in there...

**Comment by ras0219-msft** [comment]
Hi @j-forster, it looks like your issue may be different:
```
Destination must be a directory at util/copy.pl line 39
```
I'll check when I get to my machine, but if you'd like to investigate further I'd recommend starting there.

**Comment by j-forster** [comment]
A pathname with spaces made the problem.
I just moved the whole vcpkg folder to another one without spaces, and it worked perfectly.
Maybe the util/copy.pl couldn't handle that.

**Comment by samwise** [comment]
I am also unable to build openssl. I have attached the relevant logs. I am actually just looking for curl but unfortunately the curl build pull in all sorts of things (msys2? really?)



[open-ssl-build-log.tar.gz](https://github.com/Microsoft/vcpkg/files/1195556/open-ssl-build-log.tar.gz)

**Comment by ras0219-msft** [comment]
We had some trouble with the strawberry perl website over the last few days and were attempting to fix the issue.

Everything should be resolved on the latest master (d26e890b31c35e3). Could you try pulling that version and re-running install?

**Comment by samwise** [comment]
Thanks, after updating everything worked.
